### ISSUE TITLE
Rework SCA Policy for Red Hat Enterprise Linux 9

### ISSUE THREAD

**Issue by jk-olaoluwa** [wazuh/wazuh#17221]
|Wazuh version| Component | 
|---| --- |
| 4.8.0 | SCA | 
||
|Related Issue |
| #17412 |

Rework SCA Policy for Red Hat Enterprise Linux 9

- [x] Solve typos.
- [x] Increase check coverage. 
- [x] Improve rules implementation.
- [x] Improve texts.
- [x] Solve reported issues.

- #16858

**Comment by jk-olaoluwa** [comment]
https://github.com/wazuh/wazuh/issues/16858 - Solved

https://github.com/wazuh/wazuh/pull/17412/commits/977bba7f011a53ab05bad2d662ce67e66f7a4b82

**Comment by ooniagbi** [comment]
We rolled back the changes from this issue. We are reopening this to test and merge to 4.10.0.

**Comment by IsExec** [comment]
# Red Hat Enterprise Linux (RHEL) 9 SCA testing report
**Condition**: The new SCA (rework) was benchmarked against the previous RHEL 9 SCA.  

## Agent running on old SCA (RHEL9 old SCA)

###  Endpoint spec

<details>
<summary>CPU</summary>

````console
[root@agentb sca_footprint]# lscpu
Architecture:            x86_64
  CPU op-mode(s):        32-bit, 64-bit
  Address sizes:         39 bits physical, 48 bits virtual
  Byte Order:            Little Endian
CPU(s):                  2
  On-line CPU(s) list:   0,1
Vendor ID:               GenuineIntel
  Model name:            Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz
    CPU family:          6
    Model:               142
    Thread(s) per core:  1
    Core(s) per socket:  2
    Socket(s):           1
    Stepping:            10
    BogoMIPS:            4223.99
    Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx rdtscp lm c
                         onstant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq ssse3 cx16 pcid sse4_1 sse4_2 movbe popc
                         nt aes rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single pti fsgsbase bmi1 bmi2 invpcid rdseed clflushopt md_clear
                          flush_l1d arch_capabilities
Virtualization features:
  Hypervisor vendor:     KVM
  Virtualization type:   full
Caches (sum of all):
  L1d:                   64 KiB (2 instances)
  L1i:                   64 KiB (2 instances)
  L2:                    512 KiB (2 instances)
  L3:                    16 MiB (2 instances)
NUMA:
  NUMA node(s):          1
  NUMA node0 CPU(s):     0,1
````

</details>

<details>
<summary>RAM</summary>

````console
[root@agentb sca_footprint]#  free -g -h -t
               total        used        free      shared  buff/cache   available
Mem:           3.7Gi       564Mi       2.1Gi        16Mi       1.3Gi       3.2Gi
Swap:          2.0Gi          0B       2.0Gi
Total:         5.8Gi       564Mi       4.2Gi
````

</details>

<details>
<summary>Disk</summary>

````console
[root@agentb sca_footprint]# df -h
Filesystem                   Size  Used Avail Use% Mounted on
devtmpfs                     4.0M     0  4.0M   0% /dev
tmpfs                        1.9G     0  1.9G   0% /dev/shm
tmpfs                        767M   17M  750M   3% /run
/dev/mapper/rhel_rhel9-root   70G  3.0G   67G   5% /
/dev/sda1                    960M  177M  784M  19% /boot
tmpfs                        384M     0  384M   0% /run/user/1000
````
</details>


### Footprint test results
<details>
<summary>Metrics</summary>

![cpu_percent_plot](https://github.com/user-attachments/assets/245dff42-de3b-4b23-a425-6a391b3fa7a4)
![disk_percent_plot](https://github.com/user-attachments/assets/8a893495-d352-4c1c-b1f4-bed9d2a89dd1)
![disk_used_(mb)_plot](https://github.com/user-attachments/assets/ccb26e86-e3d4-42a6-a5de-7da6e3fa96f3)
![memory_percent_plot](https://github.com/user-attachments/assets/06b5ca28-6bc8-41b4-a81a-cca31f60fe97)
![memory_used_(mb)_plot](https://github.com/user-attachments/assets/15d82a99-4568-499a-aace-8afa387a5bf5)
![ProcessesCPU](https://github.com/user-attachments/assets/49f79284-a783-4942-b0f0-72060d0c5353)
![ProcessesCPU_Change_Point](https://github.com/user-attachments/assets/3cd8d917-7f6e-4863-ae93-4746fbd28321)
![ProcessesCPUlegend](https://github.com/user-attachments/assets/86663e0f-e9dd-4ed9-bba7-5b9f0f715212)
![ProcessesRAM](https://github.com/user-attachments/assets/30943d93-4371-4445-b367-1cd407f6e724)
![ProcessesRAM_Change_Point](https://github.com/user-attachments/assets/c9080fd5-44a3-4953-a329-00990f39a6ac)
![ProcessesRAMlegend](https://github.com/user-attachments/assets/a488a08b-2393-4625-9927-c747d8101e74)
![swap_percent_plot](https://github.com/user-attachments/assets/ef8e7fad-5b63-441a-a880-fbee08d0782b)
![swap_used_(mb)_plot](https://github.com/user-attachments/assets/170c9b38-b5b2-45dc-bcf2-a61536a951d4)
</details>

<details>
<summary>SCA logs</summary>

````console
[root@agentb sca_footprint]# tail /var/ossec/logs/ossec.log
2024/07/18 03:20:22 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_rhel9_linux.yml'
2024/07/18 03:20:22 sca: INFO: Security Configuration Assessment scan finished. Duration: 27 seconds.
2024/07/18 03:24:55 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/18 03:24:55 sca: INFO: Starting evaluation of policy: '/var/ossec/ruleset/sca/cis_rhel9_linux.yml'
2024/07/18 03:25:20 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_rhel9_linux.yml'
2024/07/18 03:25:20 sca: INFO: Security Configuration Assessment scan finished. Duration: 25 seconds.
2024/07/18 03:29:55 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/18 03:29:55 sca: INFO: Starting evaluation of policy: '/var/ossec/ruleset/sca/cis_rhel9_linux.yml'
2024/07/18 03:30:21 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_rhel9_linux.yml'
2024/07/18 03:30:21 sca: INFO: Security Configuration Assessment scan finished. Duration: 26 seconds.
````

</details>

[metrics.zip](https://github.com/user-attachments/files/16277447/metrics.zip)


## Agent running on new SCA (RHEL 9 rework)

### Endpoint spec
<details>
<summary>CPU</summary>

````console
[root@agenta sca_footprint]# lscpu
Architecture:            x86_64
  CPU op-mode(s):        32-bit, 64-bit
  Address sizes:         39 bits physical, 48 bits virtual
  Byte Order:            Little Endian
CPU(s):                  2
  On-line CPU(s) list:   0,1
Vendor ID:               GenuineIntel
  Model name:            Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz
    CPU family:          6
    Model:               142
    Thread(s) per core:  1
    Core(s) per socket:  2
    Socket(s):           1
    Stepping:            10
    BogoMIPS:            4223.99
    Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx rdtscp lm c
                         onstant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq ssse3 cx16 pcid sse4_1 sse4_2 movbe popc
                         nt aes rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single pti fsgsbase bmi1 bmi2 invpcid rdseed clflushopt md_clear
                          flush_l1d arch_capabilities
Virtualization features:
  Hypervisor vendor:     KVM
  Virtualization type:   full
Caches (sum of all):
  L1d:                   64 KiB (2 instances)
  L1i:                   64 KiB (2 instances)
  L2:                    512 KiB (2 instances)
  L3:                    16 MiB (2 instances)
NUMA:
  NUMA node(s):          1
  NUMA node0 CPU(s):     0,1
````

</details>

<details>
<summary>RAM</summary>

````console
[root@agenta sca_footprint]#  free -g -h -t
               total        used        free      shared  buff/cache   available
Mem:           3.7Gi       565Mi       2.1Gi        16Mi       1.3Gi       3.2Gi
Swap:          2.0Gi          0B       2.0Gi
Total:         5.8Gi       565Mi       4.2Gi
````

</details>

<details>
<summary>Disk</summary>

````console
[root@agenta sca_footprint]# df -h
Filesystem                   Size  Used Avail Use% Mounted on
devtmpfs                     4.0M     0  4.0M   0% /dev
tmpfs                        1.9G     0  1.9G   0% /dev/shm
tmpfs                        767M   17M  750M   3% /run
/dev/mapper/rhel_rhel9-root   70G  3.0G   67G   5% /
/dev/sda1                    960M  177M  784M  19% /boot
tmpfs                        384M     0  384M   0% /run/user/1000
````
</details>


### Footprint test results
<details>
<summary>Metrics</summary>

![cpu_percent_plot](https://github.com/user-attachments/assets/af007d21-27f0-4060-b59a-b3fc8d6b9dc1)
![disk_percent_plot](https://github.com/user-attachments/assets/57237e07-b5d3-48c6-a1ee-e4006a1cbbc2)
![disk_used_(mb)_plot](https://github.com/user-attachments/assets/cf4ae1e1-bdfd-454c-bdd8-7cc4f8b9efda)
![memory_percent_plot](https://github.com/user-attachments/assets/bad738dd-e821-48a0-b370-1225474e898b)
![memory_used_(mb)_plot](https://github.com/user-attachments/assets/51e82382-9013-4ade-9beb-743da2ef90ed)
![ProcessesCPU](https://github.com/user-attachments/assets/d4c0a32e-468e-4cae-91a5-d84ca59e9373)
![ProcessesCPU_Change_Point](https://github.com/user-attachments/assets/07b18a5c-8d0a-4cb1-8d83-134136b28b30)
![ProcessesCPUlegend](https://github.com/user-attachments/assets/6030b400-d76e-4432-b27e-a5f8e193ab31)
![ProcessesRAM](https://github.com/user-attachments/assets/cac77dab-b879-4ce7-a6ee-7844b99954b5)
![ProcessesRAM_Change_Point](https://github.com/user-attachments/assets/751c9cc5-26e8-4c69-98c7-8f9d112cddb2)
![ProcessesRAMlegend](https://github.com/user-attachments/assets/7621234d-7fac-4347-bae7-7525a05afd70)
![swap_percent_plot](https://github.com/user-attachments/assets/72303d18-5a20-4c21-96bb-f5882cb4b3f2)
![swap_used_(mb)_plot](https://github.com/user-attachments/assets/ba491a4c-fd26-4b75-aa72-888d77e5e596)
</details>

<details>
<summary>SCA logs</summary>

````console
[root@agenta sca_footprint]# tail /var/ossec/logs/ossec.log
2024/07/18 03:20:11 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_rhel9_linux.yml'
2024/07/18 03:20:11 sca: INFO: Security Configuration Assessment scan finished. Duration: 24 seconds.
2024/07/18 03:24:47 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/18 03:24:47 sca: INFO: Starting evaluation of policy: '/var/ossec/ruleset/sca/cis_rhel9_linux.yml'
2024/07/18 03:25:14 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_rhel9_linux.yml'
2024/07/18 03:25:14 sca: INFO: Security Configuration Assessment scan finished. Duration: 27 seconds.
2024/07/18 03:29:47 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/18 03:29:47 sca: INFO: Starting evaluation of policy: '/var/ossec/ruleset/sca/cis_rhel9_linux.yml'
2024/07/18 03:30:13 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_rhel9_linux.yml'
2024/07/18 03:30:13 sca: INFO: Security Configuration Assessment scan finished. Duration: 26 seconds.
````

</details>

[metrics.zip](https://github.com/user-attachments/files/16277508/metrics.zip)


## Analysis

  | Benchmark (Rhel   9 old SCA) | New SCA (Rhel 9   rework) | Difference
-- | -- | -- | --
CPU |   |   |  
`wazuh-modulesd` average   CPU percent | 1.85 | 1.92 | 0.07
Overall   average CPU percent during SCA runtime | 5.43 | 4.66 | -0.76
Overall peak   CPU percent during SCA runtime | 49.5 | 53 | 3.50
Memory |   |   |  
`wazuh-modulesd `average   memory (MB) | 0.02 | 0.02 | 0.00
Overall   average memory used (MB) during SCA runtime | 322.63 | 302.09 | -20.53
Overall peak   memory used peak (MB) during SCA runtime | 451.39 | 366.66 | -84.73
Disk |   |   |  
Overall   average disk used (MB) during SCA runtime | 3023.81 | 3024.76 | 0.95
Overall peak   disk used (MB) during SCA runtime | 3023.92 | 3024.85 | 0.93
Overall   average disk read bytes during SCA runtime | 733.73 | 729.45 | -4.28
Overall peak   disk read bytes during SCA runtime | 733.73 | 729.45 | -4.28
Overall   average disk write bytes during SCA runtime | 2609.90 | 2596.31 | -13.59
Overall peak   disk write bytes during SCA runtime | 2613.36 | 2602.26 | -11.10

**Comment by IsExec** [comment]
# Issue Update

Reviewing the root cause of the CPU spike

Overall peak CPU percent | 49.5 | 53 | 3.50
-- | -- | -- | --

**Steps**

- Enabled debug mode for ossec.log.
- Identify the peak CPU timestamp. 
- Compare with the timestamp on the debug logs to identify the exact process/activity/rule.

**Comment by ooniagbi** [comment]
- For check 1.9: Is there a reason why that check uses a lot of CPU? Is there an alternative way to confirm the check?
- For check 1.18.0, why is it invalid?

**Comment by IsExec** [comment]
> * For check 1.9: Is there a reason why that check uses a lot of CPU? Is there an alternative way to confirm the check?

Check 1.9 is a **manual check** that was implemented using egrep. Grep has been confirmed to consume lot of CPU and time especially when it has to go through a long list.

> For check 1.18.0, confirm if the check can be modified.

Yes, the check can be modified to include a rule that checks for the existence of the custom.conf file before the regex.

```yml
    condition: all
    rules:
      - 'f:/etc/gdm/custom.conf'
      - 'not f:/etc/gdm/custom.conf -> r:^\s*Enable\s*=\s*true'
```` 
Result
```console
This document was tested against Red Hat Enterprise Linux 9.","references":"https://www.cisecurity.org/cis-benchmarks/","passed":16,"failed":33,"invalid":0,"total_checks":49,"score":32.653060913085938,"start_time":1721392415,"end_time":1721392417,"hash":"49b80aed72648c525dccac1ad9bdb52dfbd7524f2535012f3b38bfea9d018e85","hash_file":"a87287dfbb0e486cd54c740a760577642c67cc1f086763f864e411b33f9c337e","first_scan":1}
```

**Comment by ooniagbi** [comment]
Okay. Please add a new table with these changes implemented in the SCA.

**Comment by IsExec** [comment]
# Red Hat Enterprise Linux (RHEL) 9 SCA Retesting Report
**Condition**: The new SCA (rework) was benchmarked against the previous RHEL 9 SCA.  

## Agent running on old SCA (RHEL9 old SCA)

###  Endpoint spec

<details>
<summary>CPU</summary>

````console
[root@agentb sca_footprint]# lscpu
Architecture:            x86_64
  CPU op-mode(s):        32-bit, 64-bit
  Address sizes:         39 bits physical, 48 bits virtual
  Byte Order:            Little Endian
CPU(s):                  2
  On-line CPU(s) list:   0,1
Vendor ID:               GenuineIntel
  Model name:            Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz
    CPU family:          6
    Model:               142
    Thread(s) per core:  1
    Core(s) per socket:  2
    Socket(s):           1
    Stepping:            10
    BogoMIPS:            4223.99
    Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx rdtscp lm c
                         onstant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq ssse3 cx16 pcid sse4_1 sse4_2 movbe popc
                         nt aes rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single pti fsgsbase bmi1 bmi2 invpcid rdseed clflushopt md_clear
                          flush_l1d arch_capabilities
Virtualization features:
  Hypervisor vendor:     KVM
  Virtualization type:   full
Caches (sum of all):
  L1d:                   64 KiB (2 instances)
  L1i:                   64 KiB (2 instances)
  L2:                    512 KiB (2 instances)
  L3:                    16 MiB (2 instances)
NUMA:
  NUMA node(s):          1
  NUMA node0 CPU(s):     0,1
````

</details>

<details>
<summary>RAM</summary>

````console
[root@agentb sca_footprint]# free -g -h -t
               total        used        free      shared  buff/cache   available
Mem:           3.7Gi       593Mi       2.1Gi        16Mi       1.3Gi       3.2Gi
Swap:          2.0Gi          0B       2.0Gi
Total:         5.8Gi       593Mi       4.1Gi
````

</details>

<details>
<summary>Disk</summary>

````console
[root@agentb sca_footprint]# df -h
Filesystem                   Size  Used Avail Use% Mounted on
devtmpfs                     4.0M     0  4.0M   0% /dev
tmpfs                        1.9G     0  1.9G   0% /dev/shm
tmpfs                        767M   17M  750M   3% /run
/dev/mapper/rhel_rhel9-root   70G  3.0G   67G   5% /
/dev/sda1                    960M  177M  784M  19% /boot
tmpfs                        384M     0  384M   0% /run/user/1000
````
</details>


### Footprint test results
<details>
<summary>Metrics</summary>

![cpu_percent_plot](https://github.com/user-attachments/assets/265d57ec-307c-4453-903d-cb827e94efee)
![disk_percent_plot](https://github.com/user-attachments/assets/9a4deb19-415e-4923-bfc0-d75825b129e9)
![disk_used_(mb)_plot](https://github.com/user-attachments/assets/acedf22c-cef2-4716-95de-2dbf0efbb440)
![memory_percent_plot](https://github.com/user-attachments/assets/6cd2ba61-f3b0-4efb-bf31-5d6cf0770dc7)
![memory_used_(mb)_plot](https://github.com/user-attachments/assets/2ce44a07-51fd-4bce-96dc-9ffdc960391d)
![ProcessesCPU](https://github.com/user-attachments/assets/452580d3-3c76-445f-a9a9-1e583b2f282a)
![ProcessesCPU_Change_Point](https://github.com/user-attachments/assets/072df256-82a0-4a30-be4e-8634d31d0f32)
![ProcessesCPUlegend](https://github.com/user-attachments/assets/29c56a5b-43c2-48a1-9412-ec7e8508b049)
![ProcessesRAM](https://github.com/user-attachments/assets/08e1f861-7043-40d3-9ca5-2f257b68b7ae)
![ProcessesRAM_Change_Point](https://github.com/user-attachments/assets/bc0c5015-d4db-49a7-a80f-8630b726df21)
![ProcessesRAMlegend](https://github.com/user-attachments/assets/b2aac447-f2e6-474c-a0a6-36bd82a45667)
![swap_percent_plot](https://github.com/user-attachments/assets/d56acc2b-3928-4a9a-9991-319fbe4fe8a5)
![swap_used_(mb)_plot](https://github.com/user-attachments/assets/d2ac8722-6d97-40cc-a960-23e63ec0d1e3)
</details>

<details>
<summary>SCA logs</summary>

````console
[root@agentb sca_footprint]# tail /var/ossec/logs/ossec.log
2024/07/21 23:06:38 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_rhel9_linux.yml'
2024/07/21 23:06:38 sca: INFO: Security Configuration Assessment scan finished. Duration: 10 seconds.
2024/07/21 23:11:28 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/21 23:11:28 sca: INFO: Starting evaluation of policy: '/var/ossec/ruleset/sca/cis_rhel9_linux.yml'
2024/07/21 23:11:37 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_rhel9_linux.yml'
2024/07/21 23:11:37 sca: INFO: Security Configuration Assessment scan finished. Duration: 9 seconds.
2024/07/21 23:16:28 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/21 23:16:28 sca: INFO: Starting evaluation of policy: '/var/ossec/ruleset/sca/cis_rhel9_linux.yml'
2024/07/21 23:16:37 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_rhel9_linux.yml'
2024/07/21 23:16:37 sca: INFO: Security Configuration Assessment scan finished. Duration: 9 seconds.
````

</details>

[metrics.zip](https://github.com/user-attachments/files/16326450/metrics.zip)

## Agent running on new SCA (RHEL 9 rework)

### Endpoint spec
<details>
<summary>CPU</summary>

````console
[root@agenta sca_footprint]# lscpu
Architecture:            x86_64
  CPU op-mode(s):        32-bit, 64-bit
  Address sizes:         39 bits physical, 48 bits virtual
  Byte Order:            Little Endian
CPU(s):                  2
  On-line CPU(s) list:   0,1
Vendor ID:               GenuineIntel
  Model name:            Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz
    CPU family:          6
    Model:               142
    Thread(s) per core:  1
    Core(s) per socket:  2
    Socket(s):           1
    Stepping:            10
    BogoMIPS:            4223.99
    Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx rdtscp lm c
                         onstant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq ssse3 cx16 pcid sse4_1 sse4_2 movbe popc
                         nt aes rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single pti fsgsbase bmi1 bmi2 invpcid rdseed clflushopt md_clear
                          flush_l1d arch_capabilities
Virtualization features:
  Hypervisor vendor:     KVM
  Virtualization type:   full
Caches (sum of all):
  L1d:                   64 KiB (2 instances)
  L1i:                   64 KiB (2 instances)
  L2:                    512 KiB (2 instances)
  L3:                    16 MiB (2 instances)
NUMA:
  NUMA node(s):          1
  NUMA node0 CPU(s):     0,1
````

</details>

<details>
<summary>RAM</summary>

````console
[root@agenta sca_footprint]# free -g -h -t
               total        used        free      shared  buff/cache   available
Mem:           3.7Gi       559Mi       2.1Gi        16Mi       1.3Gi       3.2Gi
Swap:          2.0Gi          0B       2.0Gi
Total:         5.8Gi       559Mi       4.2Gi
````

</details>

<details>
<summary>Disk</summary>

````console
[root@agenta sca_footprint]# df -h
Filesystem                   Size  Used Avail Use% Mounted on
devtmpfs                     4.0M     0  4.0M   0% /dev
tmpfs                        1.9G     0  1.9G   0% /dev/shm
tmpfs                        767M   17M  750M   3% /run
/dev/mapper/rhel_rhel9-root   70G  3.0G   67G   5% /
/dev/sda1                    960M  177M  784M  19% /boot
tmpfs                        384M     0  384M   0% /run/user/1000
````
</details>


### Footprint test results
<details>
<summary>Metrics</summary>

![cpu_percent_plot](https://github.com/user-attachments/assets/2af93dfb-d359-4493-a467-4f2a1333a6a9)
![disk_percent_plot](https://github.com/user-attachments/assets/b36f4709-609e-454b-b2a0-25247e7a7d82)
![disk_used_(mb)_plot](https://github.com/user-attachments/assets/44b88b99-4caa-476d-ae50-5869c63b3800)
![memory_percent_plot](https://github.com/user-attachments/assets/1cb5fb8d-96fb-49e2-b2c7-28fb56b23e01)
![memory_used_(mb)_plot](https://github.com/user-attachments/assets/22f3260c-aedd-4673-946a-e6ce59d2b2af)
![ProcessesCPU](https://github.com/user-attachments/assets/7c733f4e-4624-4673-ad87-cdffdab91859)
![ProcessesCPU_Change_Point](https://github.com/user-attachments/assets/6260c8de-729f-473e-94e6-bd2d1287c125)
![ProcessesCPUlegend](https://github.com/user-attachments/assets/24888ac4-124f-42fb-96db-5d73f8f0f166)
![ProcessesRAM](https://github.com/user-attachments/assets/69a29fa8-0a9a-4327-bf46-e75d328802d7)
![ProcessesRAM_Change_Point](https://github.com/user-attachments/assets/0bf19f73-d837-465c-8bb2-c7b00f9d2c7a)
![ProcessesRAMlegend](https://github.com/user-attachments/assets/878e4575-0c97-4041-a56f-2996a84c7cd5)
![swap_percent_plot](https://github.com/user-attachments/assets/dc9bc396-9492-4b15-8a2d-885d791462c7)
![swap_used_(mb)_plot](https://github.com/user-attachments/assets/1eddd9bf-b85b-45a5-8923-53ba437466f8)
</details>

<details>
<summary>SCA logs</summary>

````console
[root@agenta sca_footprint]# tail /var/ossec/logs/ossec.log
2024/07/21 23:08:08 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_rhel9_linux.yml'
2024/07/21 23:08:08 sca: INFO: Security Configuration Assessment scan finished. Duration: 6 seconds.
2024/07/21 23:13:02 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/21 23:13:02 sca: INFO: Starting evaluation of policy: '/var/ossec/ruleset/sca/cis_rhel9_linux.yml'
2024/07/21 23:13:08 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_rhel9_linux.yml'
2024/07/21 23:13:08 sca: INFO: Security Configuration Assessment scan finished. Duration: 6 seconds.
2024/07/21 23:18:02 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/21 23:18:02 sca: INFO: Starting evaluation of policy: '/var/ossec/ruleset/sca/cis_rhel9_linux.yml'
2024/07/21 23:18:08 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_rhel9_linux.yml'
2024/07/21 23:18:08 sca: INFO: Security Configuration Assessment scan finished. Duration: 6 seconds.
````

</details>

[metrics.zip](https://github.com/user-attachments/files/16326480/metrics.zip)

## Analysis


  | Benchmark (Rhel   9 old SCA) | New SCA (Rhel 9   rework) | Difference
-- | -- | -- | --
CPU |   |   |  
`wazuh-modulesd` average   CPU percent | 1.4 | 1.6 | 0.2
Overall   average CPU percent | 5.22 | 6.37 | 1.15
Overall peak   CPU percent | 20.6 | 18.1 | -2.5
Memory |   |   |  
`wazuh-modulesd` average   memory (MB) | 0.02 | 0.02 | 0
Overall   average memory used (MB) | 313.93 | 299.36 | -14.57
Overall peak   memory used peak (MB) | 315.52 | 299.79 | -15.73
Disk |   |   |  
Overall   average disk used (MB) | 3031.6 | 3030.96 | -0.64
Overall peak   disk used (MB) | 3031.61 | 3030.98 | -0.63
Overall   average disk read bytes | 737.36 | 739.29 | 1.93
Overall peak   disk read bytes | 737.36 | 739.29 | 1.93
Overall   average disk write bytes | 3539.27 | 3393.11 | -146.16
Overall peak   disk write bytes | 3540.53 | 3393.88 | -146.65
  |   |   |

**Comment by ooniagbi** [comment]
LGTM!

**Comment by ooniagbi** [comment]
https://github.com/wazuh/wazuh/pull/24667 merged to 4.10.0
